### PR TITLE
Finally remove INT8x4 support.

### PR DESCRIPTION
**Pull request by atamazov**
- Completes #2441 ([why we want to use two PRs](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2441#issuecomment-1753927378))
  - The exception is the driver, where I kept the INT8x4 related code for illustrative purposes
- Resolves [postponed review comments from #2441](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2441#pullrequestreview-1668434374)
- Fixes some [minor bug from #2251](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2251/files#r1359640509)
- Promotes fin submodule to the results of merging of https://github.com/ROCmSoftwarePlatform/MIFin/pull/89.

----
- ~⚠️ Should not be merged in until https://github.com/ROCmSoftwarePlatform/MIFin/pull/89 is done.~
- ~⚠️ This PR contains some unwanted changes (it temporarily switches fin submodule to another repo & branch) that necessary to pass CI while https://github.com/ROCmSoftwarePlatform/MIFin/pull/89 is not merged in. These will be reverted prior merge.~

### CODE REVIEW

**Comment by atamazov** [comment]
@JehandadKhan @amberhassaan @averinevg @junliume Fin promoted to the most recent develop. Description updated. Ready for final review, CI testing & merging.

**Comment by atamazov** [comment]
@junliume From my POV, https://github.com/ROCmSoftwarePlatform/MIOpen/labels/ON_HOLD can be removed .

**Review by CAHEK7 — APPROVED** [review]

**Review by amberhassaan — APPROVED** [review]
Looks good.

File: src/include/miopen/datatype.hpp [hunk 1]
```diff
@@ -180,7 +173,7 @@ inline KernelBuildParameters GetDataTypeKBP(miopenDataType_t type)
     if(use_fp8 != 0)
         kbp.Define("MIOPEN_USE_FP8", use_fp8);
     if(use_bfp8 != 0)
-        kbp.Define("MIOPEN_USE_FP8", use_bfp8);
+        kbp.Define("MIOPEN_USE_BFP8", use_bfp8);
```
Comment (JehandadKhan) [inline src/include/miopen/datatype.hpp:176]: Good catch! @cderb  Please note!

**Review by JehandadKhan — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Review by averinevg — APPROVED** [review]

**Pull request merged**